### ISSUE TITLE
Annotation on inner class constructor cause error

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#275]
```
What steps will reproduce the problem?
1.

TestInner.java:

import checkers.tainting.quals.*;

class Test {
        class Inner {
                @Tainted Inner() { } 
        }
}

./javac -processor checkers.tainting.TaintingChecker TestInner.java

What is the expected output? What do you see instead?
Expected no errors. Received:

error: TypeFromElement.getLocationTypeATM: only declared types and arrays can 
have annotations with location; found type: void location: INNER_TYPE; invoke 
the compiler with -AprintErrorStack to see the stack trace.
1 error


What version of the product are you using? On what operating system?
Checker framework changeset: 7223:969c9cf4c550

Linux 3.9.9-0cse.0 #1 SMP Tue Jul 9 14:16:52 PDT 2013 x86_64 x86_64 x86_64 
GNU/Linux

Please provide any additional information below.
```

Original issue reported on code.google.com by `mcart...@cs.washington.edu` on 18 Nov 2013 at 5:10
